src/libsyntax/ast.rs

```diff
@@ -112,6 +113,15 @@ impl PartialEq<Symbol> for Path {
     }
 }
 
+impl<CTX> HashStable<CTX> for Path {
+    fn hash_stable(&self, hcx: &mut CTX, hasher: &mut StableHasher) {
+        self.segments.len().hash_stable(hcx, hasher);
+        for segment in &self.segments {
+            segment.ident.name.hash_stable(hcx, hasher);
```

Comment:
@michaelwoerister This skips all the other fields in `PathSegment` and also the span field in `Path`. Is this intentional?

Reply:
It looks like the NodeId field was added later and the whole implementation was lifted to a more general context in https://github.com/rust-lang/rust/commit/759bd01e039452a1a357d347aea51348f9ffc443. It would be good to hash the generic arguments too.